dotnet/samples/Demos/AgentWithHistoryTruncation/TruncatingChatChannel.cs

```diff
@@ -0,0 +1,51 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+using System.Runtime.CompilerServices;
+using Microsoft.SemanticKernel;
+using Microsoft.SemanticKernel.Agents;
+using Microsoft.SemanticKernel.ChatCompletion;
+
+namespace AgentWithHistoryTruncation;
+
+internal class TruncatingChatChannel : ChatHistoryChannel
```

Comment:
All of this can simply be integrated into `ChatHistoryChannel` and `ChatHistoryKernelAgent`